### PR TITLE
Add Discord RPC

### PR DESCRIPTION
**Pull request by TabulateJarl8**
This pull request adds Discord rich presence to Legendary. This does unfortunately change the behavior of Legendary. Instead of releasing the game process, the process now blocks because the Discord rich presence needs to continue running while the game is running. I attempted to just track the PID of the game but the original process seems to start a bunch of child processes and then end itself, which doesn't let me track it. If someone does merge this, you might want to make your own Discord app for Legendary on [https://discord.com/developers/applications](https://discord.com/developers/applications), and use that client ID instead of the one I currently have. I've made 3 different versions of the logo for the presence, and I'll attach them to the PR so you can choose which you like best. I've also attached an example of the rich presence. Tell me if you need any more info.

image
Presence images: [https://imgur.com/a/4zDDs4R](https://imgur.com/a/4zDDs4R) (I'm currently using the middle one with the transparent background)

### CODE REVIEW

**Comment by Dummerle** [comment]
This looks awesome. May I use it in github.com/Dummerle/Rare?

**Comment by TabulateJarl8** [comment]
Yeah, I have no problem with you using it. If legendary merges this PR, it may break something in your project though. Then again it could just be removed if they add it. Whatever you choose to do is fine

File: legendary/cli.py [hunk 1]
```diff
@@ -510,7 +513,38 @@ def launch_game(self, args, extra):
...
+            game_process.wait()
+
+            if not args.disable_rpc:
+                ns.finish_rpc = True
+                rpc.join()
+
+
+    def startRPC(self, app_name, ns):
+        try:
+            RPC = Presence('828711025863688192')
```
Comment (adityaruplaha) [inline legendary/cli.py:535]: Everything looks good, but maybe add a comment explaining this magic number? I assume this is for connecting to Discord.

**Review by adityaruplaha — APPROVED** [review]
Thanks for this, Heroic also got a feature request. About the blocking process, I'm sure we can fix it on our side.

Also is Legendary still non-blocking with `--disable-rpc`?

**Comment by TabulateJarl8** [comment]
> Thanks for this, Heroic also got a feature request. About the blocking process, I'm sure we can fix it on our side.
> 
> Also is Legendary still non-blocking with `--disable-rpc`?

I made it so that Legendary is blocking with `--disable-rpc` for consistency reasons, but if you would like I could make it non blocking very easily.

File: legendary/cli.py (hunk 1 above)
Comment (TabulateJarl8) [reply legendary/cli.py:540]: I've now commented all of my code

**Comment by adityaruplaha** [comment]
I would prefer it to be blocking in all cases instead of being inconsistent really.

File: legendary/cli.py (hunk 1 above)
Comment (adityaruplaha) [reply legendary/cli.py:540]: Yep, LGTM.

**Comment by TabulateJarl8** [comment]
Ok, I won't change it to be non-blocking in that case

**Comment by derrod** [comment]
Thanks, but this is not something I think is appropiate for a command line application. Certainly not the way it's currently designed and intended to function. A future GUI may have use for Discord RPC however since that's inteded to remain running after application launch anyway.

**Pull request closed without merging**